data-collection/src/main/java/refactoringml/App.java

```diff
@@ -74,7 +74,13 @@ public App (String datasetName,
 		this.storeFullSourceCode = storeFullSourceCode;
 	}
 
+
 	public Project run () throws Exception {
+		// do not run if the project is already in the database
+		if (db.projectExists(gitUrl)) {
+			String message = String.format("Project %s already in the database", gitUrl);
+			throw new IllegalArgumentException(message);
```

Comment:
Maybe a `log.error` message, so that we can later grep it in the logs?

Reply:
Forget it, just noticed this exception will pop up to the Run class, and it will be logged there.